lang/csharp/src/apache/main/Reflect/ClassCache.cs

```diff
@@ -254,6 +254,16 @@ public void LoadClassCache(Type objType, Schema s)
                     break;
                 case NamedSchema ns:
                     EnumCache.AddEnumNameMapItem(ns, objType);
+                    break;
+                case UnionSchema us:
+                    foreach (var o in us.Schemas)
```

Comment:
Please correct me if I'm wrong, but I think we can only support certain types of unions:
1. Those with only 1 branch
1. Those with only 2 branches, one of which is `"null"`

If that's true, I think we should make this `case` statement more restrictive. For example, throwing an exception when it doesn't meet these criteria.

Reply:
Added the test. Multiple branches in a union are supported but when a record type is first defined in a union and the union is not of type ["null", "recordType"] then the type needs to be manually registered in the class cache. Added a test to (I hope) make this clearer.  Also added to the README.